Framework/Algorithms/src/MaskNonOverlappingBins.cpp

```diff
@@ -0,0 +1,136 @@
+#include "MantidAlgorithms/MaskNonOverlappingBins.h"
+
+#include "MantidAPI/MatrixWorkspace.h"
+#include "MantidKernel/MultiThreaded.h"
+
+namespace {
+/// Constants for the algorithm's property names.
+namespace Prop {
+std::string const COMPARISON_WS{"ComparisonWorkspace"};
+std::string const INPUT_WS{"InputWorkspace"};
+std::string const MASK_PARTIAL{"MaskPartiallyOverlapping"};
+std::string const OUTPUT_WS{"OutputWorkspace"};
+} // namespace Prop
+} // namespace
+
+namespace Mantid {
+namespace Algorithms {
+// Mantid Repository : https://github.com/mantidproject/mantid
+//
+// Copyright &copy; 2018 ISIS Rutherford Appleton Laboratory UKRI,
+//     NScD Oak Ridge National Laboratory, European Spallation Source
+//     & Institut Laue - Langevin
+// SPDX - License - Identifier: GPL - 3.0 +
+
+DECLARE_ALGORITHM(MaskNonOverlappingBins)
+
+/// Algorithms name for identification. @see Algorithm::name
+std::string const MaskNonOverlappingBins::name() const {
+  return "MaskNonOverlappingBins";
+}
+
+/// Algorithm's version for identification. @see Algorithm::version
+int MaskNonOverlappingBins::version() const { return 1; }
+
+/// Algorithm's category for identification. @see Algorithm::category
+std::string const MaskNonOverlappingBins::category() const {
+  return "Transforms\\Masking";
+}
+
+/// Algorithm's summary for use in the GUI and help. @see Algorithm::summary
+std::string const MaskNonOverlappingBins::summary() const {
+  return "Marks bins in InputWorkspace which are out of the X range of the "
+         "second workspace.";
+}
+
+/// Return a list of the names of related algorithms.
+std::vector<std::string> const MaskNonOverlappingBins::seeAlso() const {
+  return {"MaskBins", "MaskBinsIf"};
```

Comment:
👍 for the `seeAlso`. Do you also want to add to the `seeAlso` for the algorithms you mention here? Form a two way link.

Reply:
In fact, MaskBinsIf just a pull request #23751 yet.